components/angular-tomitribe-select/src/tomitribe-select.ts

```diff
@@ -178,6 +179,22 @@ module tomitribe_select {
         }
     }
 
+    function tribeSelectSaveSearch() {
+        return {
+            require: 'uiSelect',
+            replace: false,
+            link: link
+        };
+
+        function link(scope, element, attrs, uiSelect) {
+            uiSelect.resetSearchInput = false;
+
+            scope.$on('uis:select', function(arg) {
+                uiSelect.search = uiSelect.selected;
```

Comment:
Please include possibility of selected item to be an object, you will need fill attribute.